anaconda_project/internal/default_conda_manager.py

```diff
@@ -105,22 +179,59 @@ def _write_timestamp_file(self, prefix, spec):
             # fail we will survive
             pass
 
-    def _find_conda_missing(self, prefix, spec):
+    def resolve_dependencies(self, package_specs, channels, platforms):
+        by_platform = {}
+
+        # This has no reason to be a loop now, but it will
+        # soon when we do multi-platform resolve
+        current = conda_api.current_platform()
+        resolve_for_platforms = list(platforms)
+        # always resolve "current" first because it's confusing if
+        # an error says resolution failed on another platform when
+        # the real issue is that resolution will fail on all platforms.
+        if current in resolve_for_platforms:
+            resolve_for_platforms.remove(current)
+            resolve_for_platforms = [current] + resolve_for_platforms
+        for conda_platform in resolve_for_platforms:
+            try:
+                deps = conda_api.resolve_dependencies(pkgs=package_specs, platform=conda_platform, channels=channels)
+            except conda_api.CondaError as e:
+                raise CondaManagerError("Error resolving for {}: {}".format(conda_platform, str(e)))
+            locked_specs = ["%s=%s=%s" % dep for dep in deps]
+            by_platform[conda_platform] = sorted(locked_specs)
+
+        by_platform = _extract_common(by_platform)
+
+        lock_set = CondaLockSet(package_specs_by_platform=by_platform, platforms=resolve_for_platforms)
+        return lock_set
+
+    def _find_conda_deviations(self, prefix, env_spec):
         try:
             installed = conda_api.installed(prefix)
         except conda_api.CondaError as e:
             raise CondaManagerError("Conda failed while listing installed packages in %s: %s" % (prefix, str(e)))
 
-        # TODO: we don't verify that the environment contains the right versions
-        # https://github.com/Anaconda-Server/anaconda-project/issues/77
-
         missing = set()
+        wrong_version = set()
+
+        for spec_string in env_spec.conda_packages_for_create:
+            spec = conda_api.parse_spec(spec_string)
+            name = spec.name
 
-        for name in spec.conda_package_names_set:
             if name not in installed:
                 missing.add(name)
-
-        return sorted(list(missing))
+            else:
+                # The only constraint we are smart enough to understand is
+                # the one we put in the lock file, which is plain =.
+                # We can't do version comparisons, which is a bug.
```

Comment:
In conda, `jinja2=2.8` is not a strict/exact version match. `jinja2=2.8` matches any sub-version of jinja 2.8, e.g. `jinja2 2.8.1`.

The only *actual* strict matches include the build strings, for example: `jinja2=2.8.1=py35_0`

This causes `anaconda-project` to keep trying to correct this discrepancy (which is failing for unrelated reasons).